oneflow/core/ep/cpu/cpu_device_manager.cpp

```diff
@@ -28,7 +31,16 @@ DeviceManagerRegistry* CpuDeviceManager::registry() const { return registry_; }
 
 std::shared_ptr<Device> CpuDeviceManager::GetDevice(size_t device_index) {
   std::lock_guard<std::mutex> lock(device_mutex_);
-  if (!device_) { device_.reset(new CpuDevice(this)); }
+  if (!device_) {
+    CpuDevice* cpu_device = new CpuDevice(this);
```

Comment:
```suggestion
    device_.reset(new CpuDevice(this));
```

下面直接操作 device_ 吧，尽量少的暴露裸指针

Reply:
NOTE：这样改不太方便。仍然使用原写法吧

Reply:
device_.reset(new CpuDevice(this));
    dynamic_cast<CpuDevice*>(device_.get())->SetNumParallels(computing_cores);